guide_drafts/trait_derives.md

```diff
@@ -0,0 +1,954 @@
+Deriving Traits in Depth
+===============
+
+Part of what makes Diesel's query builder so powerful is
+its ability to assist writing safe SQL queries in Rust.
+Diesel enables this level of safety through implementing 
+a series of traits on your structs.
+Writing these implementations by hand can be very laborious,
+so Diesel offers custom derives.
+
+In this guide,
+we will cover each trait in detail in terms of its use cases 
+and usage considerations.
+To be able to use these derives,
+make sure you have `#[macro_use] extern crate diesel;` at the root of your project.
+
+Throughout this guide,
+we will be looking at examples for each trait and how they interact with each other.
+Some of the example code will be implementing basic [CRUD] database operations.
+We'll be covering creating, reading, and updating data.
+The details of those operations will be not be covered beyond their relevance to the demonstrated trait.
+
+In general, it may be more helpful to think of Diesel as a SQL query builder.
+While Diesel does offer some standard ORM (Object Relation Mapper) features,
+Diesel's code generation derives are for safely building SQL queries.
+
+[CRUD]: https://en.wikipedia.org/wiki/Create,_read,_update_and_delete
+
+- [Queryable](#queryable)
+- [QueryableByName](#queryablebyname)
+- [Insertable](#insertable)
+- [Identifiable](#identifiable)
+- [AsChangeset](#aschangeset)
+- [Associations](#associations)
+
+## Queryable
+A `Queryable` struct is one that represents the 
+data structure returned from a database query.
```

Comment:
> represents the data structure

Hmm. Structs _are_ data structures. (They can also represent data structures by representing an AST but that's an edge case…) You can also create a struct that has a Queryable impl manually… so 'represent' may be a bit strong.

Maybe go with "struct is one that can be creating from the result of a database query"?